rat/variables.py

```diff
@@ -148,6 +164,7 @@ class IndexUse:
     shifts: Tuple[Union[str, None]] = None
 
     def to_numpy(self):
+        self.df = self.df.loc[:, self.names]
```

Comment:
Any memory of where this came from?

Reply:
So `variable.Index.base_df` is the "reference dataframe", which contains the unique subscripts sets and assigns an index to each of them. In `variable.IndexUse`, we receive some other subscript dataframe, which contains subscripts that are being used in that row. `to_numpy()` would get the reference indexes of the subscript dataframe.

That line in particular, say we have `skills[team, year] ~ normal(skills_mu[year], tau);`. 
It takes the index dataframe of `skills_mu`(`variable.IndexUse.df` of `skills_mu`), which would be defined by the reference dataframe of `skills`(`Index.base_df` of `skills`) and extract only the column that `skills_mu` use, which would be `year`.

Once that's done, it compares that to the reference dataframe of itself, yielding numpy indexes of the unique subscripts that are within the index dataframe.